File: targets/java-web/src/main/java/com/example/vulnapp/vulns/Vuln17InsecureCookie.java
package com.example.vulnapp.vulns;

import com.example.vulnapp.vulns.crypto.Secrets;
import org.springframework.web.bind.annotation.*;
import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletResponse;
import java.util.*;

@RestController
public class Vuln17InsecureCookie {
    @PostMapping("/vuln17/login")
    public Map<String,Object> login(@RequestParam String username, HttpServletResponse response) {
        String token = Secrets.weakToken(username.hashCode());
        // VULN:VULN-17:CWE-614:config auth cookie set without HttpOnly/Secure flags
        Cookie c = new Cookie("auth", token);
        c.setPath("/");
        response.addCookie(c);
        Map<String,Object> out = new HashMap<>();
        out.put("status","ok"); out.put("token", token);
        return out;
    }
}

File: targets/kotlin-web/src/main/kotlin/com/example/vulnapp/vulns/Vuln17InsecureCookie.kt
package com.example.vulnapp.vulns

import org.springframework.web.bind.annotation.*
import javax.servlet.http.HttpServletResponse

@RestController
class Vuln17InsecureCookie {

    @PostMapping("/vuln17/login")
    fun login(@RequestParam username: String, response: javax.servlet.http.HttpServletResponse): Map<String, Any?> {
        val token = com.example.vulnapp.vulns.crypto.Secrets.weakToken(username.hashCode().toLong())
        // VULN:VULN-17:CWE-614:config auth cookie set without HttpOnly/Secure flags
        val c = javax.servlet.http.Cookie("auth", token); c.path = "/"; response.addCookie(c)
        return mapOf("status" to "ok", "token" to token)
    }
}
